### PR TITLE
Adding daos addresses to labels

### PR DESCRIPTION
**Pull request by henrystats**
Brief comments on the purpose of your changes:

Hola ser @soispoke, here's the pr to include daos addresses to the labels table. 

**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributors)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/labels/daos/labels_daos.sql [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    dao as address, 
+    CASE 
+        WHEN dao_creator_tool = 'aragon' THEN 'DAOs: Aragon'
+        WHEN dao_creator_tool = 'colony' THEN 'DAOs: Colony'
+        WHEN dao_creator_tool = 'dao-haus' THEN 'DAOs: DAO Haus'
+        WHEN dao_creator_tool = 'syndicate' THEN 'DAOs: Syndicate Investment Club'
+    END as name,
+    'daos' as category,
+    'henrystats' as contributor,
+    'daos-addresses-spell' as source,
```
Comment (soispoke) [inline models/labels/daos/labels_daos.sql:14]: source should be either `query` or `static` (in this case, query) :)

File: models/labels/daos/labels_daos.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    array(blockchain) as blockchain,
+    dao_wallet_address as address, 
+    CASE 
+        WHEN dao_creator_tool = 'aragon' THEN 'DAOs: Aragon'
+        WHEN dao_creator_tool = 'colony' THEN 'DAOs: Colony'
+        WHEN dao_creator_tool = 'dao-haus' THEN 'DAOs: DAO Haus' 
+    END as name,
+    'daos' as category,
+    'henrystats' as contributor,
+    'daos-addresses-spell' as source,
```
Comment (soispoke) [inline models/labels/daos/labels_daos.sql:33]: same comment here, should be source should be `query`

File: models/labels/daos/labels_daos_schema.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+version: 2
+
+models:
+  - name: labels_daos
```
Comment (soispoke) [inline models/labels/daos/labels_daos_schema.yml:4]: I think adding a uniqueness test to make sure there are no duplicates would be great

**Review by soispoke — COMMENTED** [review]
thanks @henrystats ! Made a few comments/suggestions

Also, I know I probably should have said that before but I really think we should rename `daos` to `dao` to match our naming conventions (e.g., nft.trades, dex.trades, etc...)

I can definitely do it and tag you in the PR but if you can take this one it would be great as well, lmk 👍

File: models/labels/daos/labels_daos_schema.yml (hunk 3 above)
Comment (henrystats) [reply models/labels/daos/labels_daos_schema.yml:4]: Thank you ser @soispoke, I'll update and add tests and also change the query source to `query`. 

Also, I can definitely do the pr to update and change from `daos` to `dao`, I have another pr open right now (to add polygon wallets & transactions) and I'm using `daos`. 

Should I submit that **pr** after these pr's are merged? or should I submit a name change to all the models along with this pr? and update my other pr as well?

**Comment by soispoke** [comment]
You can do whatever is easier for you @henrystats :) Thanks for taking this on !

**Comment by henrystats** [comment]
okaie ser, I guess it's better to make the changes now 😅 once and for all. I'll get to it shortly and let you know once I'm done. Thank you very much for your help!

**Comment by henrystats** [comment]
Unique tests added, source changed to query and name changed to DAO.

This is now ready for review/merging ser @soispoke !

File: models/labels/dao/labels_dao.sql [hunk 4]
```diff
@@ -0,0 +1,45 @@
+{{config(alias='dao')}}
+
+SELECT 
+    array(blockchain) as blockchain,
+    dao as address,
```
Comment (soispoke) [inline models/labels/dao/labels_dao.sql:5]: qq @henrystats I'm not sure I understand the difference between the `dao` and `dao_wallet_address` columns here ?

Comment (henrystats) [reply models/labels/dao/labels_dao.sql:5]: hola @soispoke, so, for example when you create a DAO using Aragon, you also need to create an APP (a smart contract) that would receive and send funds that belong to that DAO and for other actions too. 

For DAO Haus, the moloch (the DAO) receives and sends funds, but a moloch can also set up a minion (a gnosis safe with reality.eth module) that can be used by the DAO to receive and send funds.

For colony, it's the DAO address that receives and sends fund. So on the dao.addresses table, there are multiple entries of **DAO** but the dao wallet address table is different (daos can have multiple wallets, e.g with Aragon).

That's why I'm selecting both dao & dao_wallet_address since they belong to a DAO.

**Comment by soispoke** [comment]
Sounds good thanks for taking the time to give context @henrystats ! lgtm

**Review by soispoke — APPROVED** [review]
